test/component/profile_page_test.dart

```diff
@@ -49,96 +52,182 @@ void main() {
-    ];
-
-    Widget profilePageWidget = ProviderScope(
-      overrides: firebaseAuthMocksOverrides,
-      child: const MaterialApp(
-        onGenerateRoute: generateRoute,
-        title: "Profile page",
-        home: ProfilePage(),
-      ),
-    );
-
-    await tester.pumpWidget(profilePageWidget);
-    await tester.pumpAndSettle();
-
-    // Check that the post tab is displayed
-    final postTab = find.byKey(ProfilePage.postTabKey);
-    expect(postTab, findsOneWidget);
-
-    // Check that the comment tab is displayed
-    final commentTab = find.byKey(ProfilePage.commentTabKey);
-    expect(commentTab, findsOneWidget);
-
-    //Check that post column is displayed
-    final postColumn = find.byKey(ProfilePage.postColumnKey);
-    expect(postColumn, findsOneWidget);
-
-    // Tap on the comment tab
-    await tester.tap(commentTab);
-    await tester.pumpAndSettle();
-
-    // Check that the comment column is displayed
-    final commentColumn = find.byKey(ProfilePage.commentColumnKey);
-    expect(commentColumn, findsOneWidget);
+  group("Profile widget page testing", () {
+    setUp(() async {
+      setupFirebaseAuthMocks();
+    });
+
+    ProviderScope getMockedProxima() {
+      MockFirebaseAuth auth = MockFirebaseAuth(signedIn: true);
+
+      final firebaseAuthMocksOverrides = [
+        firebaseAuthProvider.overrideWithValue(auth),
+      ];
+
+      return ProviderScope(
+        overrides: firebaseAuthMocksOverrides,
+        child: const MaterialApp(
+          onGenerateRoute: generateRoute,
+          title: "Profile page",
+          home: ProfilePage(),
+        ),
+      );
+    }
+
+    testWidgets("Various widget are displayed", (tester) async {
+      await tester.pumpWidget(getMockedProxima());
+      await tester.pumpAndSettle();
+
+      // Check that badges are displayed
+      final badgeCard = find.byKey(InfoCardBadge.infoCardBadgeKey);
+      expect(badgeCard, findsWidgets);
+
+      //Check that the post card is displayed
+      final postCard = find.byKey(InfoCardPost.infoCardPostKey);
+      expect(postCard, findsWidgets);
+
+      // Check that the info column is displayed
+      final infoColumn = find.byKey(ProfilePage.postColumnKey);
+      expect(infoColumn, findsOneWidget);
+
+      // Check that the info row is displayed
+      final infoRowWidget = find.byKey(InfoRow.infoRowKey);
+      expect(infoRowWidget, findsOneWidget);
+
+      //Check that centauri points are displayed
+      final centauriPoints = find.byKey(UserAccount.centauriPointsKey);
+      expect(centauriPoints, findsOneWidget);
+
+      //Check that the user account is displayed
+      final userAccount = find.byKey(UserAccount.userInfoKey);
+      expect(userAccount, findsOneWidget);
+
+      //Check that the tab is displayed
+      final tab = find.byKey(ProfilePage.tabKey);
+      expect(tab, findsOneWidget);
+    });
+
+    testWidgets("Tab working as expected", (tester) async {
```

Comment:
There are two testWidgets with name "Tab working as expected".

Reply:
Good catch ! Solved in b15b0a9